apps/dashboard/app/helpers/recently_used_apps_helper.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+# Helper methods for RecentlyUsedApps widget
+module RecentlyUsedAppsHelper
+  # Returns the cached value for recently used apps
+  def recently_used_apps
+    Rails.cache.fetch('recently_used_apps', expires_in: 1.hour) do
+      load_recently_used_apps
+    end
+  end
+
+  private
+
+  # Returns the 4 most recently used interactive applications based on their cache file
+  # @return [Array<BatchConnect::App>] The 4 most recently used apps
+  def load_recently_used_apps
+    cache_files = BatchConnect::Session.cache_root.children.sort_by do |pathname|
+                    File.mtime(pathname)
+                  end.reverse.map do |pathname|
+                    pathname.basename.to_s
+                  end.slice(0, 4)
+    sys_apps_index = {}
+    SysRouter.apps.select(&:batch_connect_app?).each do |ood_app|
```

Comment:
```suggestion
   (SysRouter.apps + DevRouter.apps + UsrRouter.apps).select(&:batch_connect_app?).each do |ood_app|
```

We may as well add this here and now. I'm not super concerned about `dev` apps, but there are sites that use `usr` apps almost exclusively. That is, they don't really even have `sys` apps, they just share all their apps from the staff and/or regular users.

Reply:
Done. Updated to use `@sys_apps + @dev_apps + @usr_apps` that are initialized in the `ApplicationController` class.

Reply:
That'll work for now.  IIRC member variables are generated every page load so caching the member variables is a little funny, but that's OK (let's hope so anyhow).